jina/proto/jina.proto

```diff
@@ -53,7 +53,7 @@ message SparseNdArray {
     DenseNdArray values = 2;
 
     // A 1-D int64 tensor of shape [ndims], which specifies the dense_shape of the sparse tensor.
-    repeated uint64 dense_shape = 3;
+    repeated int64 dense_shape = 3;
```

Comment:
why not uint?

Reply:
i had the same question, but after looking at all references i have onnx, tf, they all use int64

Reply:
my guess is that `int64` is more natural to join the computation when deploying on multi-arch machines, cpu, gpu, embedded device.

Reply:
so more for being conservative?

Reply:
or that we suspect they must have their own reasons and better be conservative?

Reply:
my experience is that `shape` sometimes join the computation flow of those frameworks, explicitly casting probably better.

Reply:
ok did not see second message yet